src/singletons/UsernameColors.cpp

```diff
@@ -0,0 +1,44 @@
+#include "UsernameColors.hpp"
+
+namespace chatterino {
+UsernameColors &UsernameColors::getInstance()
+{
+    static UsernameColors *instance = new UsernameColors();
+    return *instance;
+}
+
+QColor UsernameColors::getColor(const int userId)
+{
+    static const std::vector<QColor> twitchUsernameColors{
+        QColor(255, 0, 0), // Red
+        QColor(0, 0, 255), // Blue
+        QColor(0, 255, 0), // Green
+        QColor(178, 34, 34), // FireBrick
+        QColor(255, 127, 80), // Coral
+        QColor(154, 205, 50), // YellowGreen
+        QColor(255, 69, 0), // OrangeRed
+        QColor(46, 139, 87), // SeaGreen
+        QColor(218, 165, 32), // GoldenRod
+        QColor(210, 105, 30), // Chocolate
+        QColor(95, 158, 160), // CadetBlue
+        QColor(30, 144, 255), // DodgerBlue
+        QColor(255, 105, 180), // HotPink
+        QColor(138, 43, 226), // BlueViolet
+        QColor(0, 255, 127) // SpringGreen
+    };
+
+    const auto it = this->usernameRandomColors_.find(userId);
+    if (it == this->usernameRandomColors_.cend())
+    {
+        const auto color = twitchUsernameColors[
```

Comment:
I would prefer if colors weren't fully randomized, rather deterministically chosen, so a user will have the same color all the time even after restarts. this could probably be done using standard C++ <random>

Reply:
Should I store the `usernameRandomColors_` map in a separated file?

Reply:
It should be randomized using a deterministic method (mersienne twister) on the user id so the calling randomColor on the same user id with always produce the same color

Reply:
Okay, now I get it, thanks.